lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -344,14 +369,21 @@ defmodule Kernel.ParallelCompiler do
     :ok
   end
 
-  defp maybe_check_modules(result, runtime_modules, schedulers) do
+  defp maybe_check_modules(result, runtime_modules, state) do
+    %{schedulers: schedulers, profile: profile} = state
+
     if :elixir_config.get(:bootstrap) do
       binaries = for {{:module, module}, {binary, _map}} <- result, do: {module, binary}
       {binaries, []}
     else
       compiled_modules = checker_compiled_modules(result)
       runtime_modules = checker_runtime_modules(runtime_modules)
-      Module.ParallelChecker.verify(compiled_modules, runtime_modules, schedulers)
+      num_modules = length(compiled_modules) + length(runtime_modules)
+      message = "Finished group pass check of #{num_modules} modules in ${time}ms"
```

Comment:
maybe move the message inside `maybe_profile` and compute the length only in there as well?